Plugin/Framework/Search/Request/AbstractHandler.php

```diff
@@ -0,0 +1,308 @@
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * @author Nosto Solutions Ltd <contact@nosto.com>
+ * @copyright 2020 Nosto Solutions Ltd
+ * @license http://opensource.org/licenses/BSD-3-Clause BSD 3-Clause
+ *
+ */
+
+namespace Nosto\Cmp\Plugin\Framework\Search\Request;
+
+use Magento\Store\Model\StoreManagerInterface;
+use Nosto\Cmp\Helper\SearchEngine;
+use Nosto\Cmp\Logger\LoggerInterface;
+use Nosto\Cmp\Model\Filter\FiltersInterface;
+use Nosto\Cmp\Model\Service\Recommendation\StateAwareCategoryServiceInterface;
+use Nosto\Cmp\Plugin\Catalog\Block\ParameterResolverInterface;
+use Nosto\Cmp\Utils\CategoryMerchandising;
+use Nosto\Cmp\Utils\Search;
+use Nosto\Tagging\Helper\Account as NostoHelperAccount;
+
+abstract class AbstractHandler
+{
+    const KEY_ES_PRODUCT_ID = '_id';
+    const KEY_MYSQL_PRODUCT_ID = 'entity_id';
+    const KEY_BIND_TO_QUERY = 'catalog_view_container';
+    const KEY_BIND_TO_GRAPHQL = 'graphql_product_search';
+    const KEY_CATEGORY_FILTER = 'category_filter';
+    const KEY_QUERIES = 'queries';
+    const KEY_FILTERS = 'filters';
+    const KEY_VALUE = 'value';
+    const KEY_CMP = 'nosto_cmp_id_search';
+    const KEY_RESULTS_FROM = 'from';
+    const KEY_RESULT_SIZE = 'size';
+
+    public static $nostoTmpSort = [5, 11, 401, 2023, 1];
+
+    /**
+     * @var ParameterResolverInterface
+     */
+    private $parameterResolver;
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * @var SearchEngine
+     */
+    private $searchEngineHelper;
+
+    /**
+     * @var StoreManagerInterface
+     */
+    protected $storeManager;
+
+    /**
+     * @var NostoHelperAccount
+     */
+    private $accountHelper;
+
+    /**
+     * @var StateAwareCategoryServiceInterface
+     */
+    protected $categoryService;
+
+    /**
+     * @param ParameterResolverInterface $parameterResolver
+     * @param SearchEngine $searchEngineHelper
+     * @param StoreManagerInterface $storeManager
+     * @param NostoHelperAccount $nostoHelperAccount
+     * @param StateAwareCategoryServiceInterface $categoryService
+     * @param LoggerInterface $logger
+     */
+    public function __construct(
+        ParameterResolverInterface $parameterResolver,
+        SearchEngine $searchEngineHelper,
+        StoreManagerInterface $storeManager,
+        NostoHelperAccount $nostoHelperAccount,
+        StateAwareCategoryServiceInterface $categoryService,
+        LoggerInterface $logger
+    ) {
+        $this->parameterResolver = $parameterResolver;
+        $this->logger = $logger;
+        $this->searchEngineHelper = $searchEngineHelper;
+        $this->storeManager = $storeManager;
+        $this->accountHelper = $nostoHelperAccount;
+        $this->categoryService = $categoryService;
+    }
+
+    /**
+     * @param array $requestData
+     * @return array
+     */
+    public function handle(array $requestData)
```

Comment:
The data is returned at the end.

Reply:
This is still not working. You're not assigning the returned variable here: https://github.com/Nosto/nosto-magento2-cmp/blob/feature/refactor-request-cleaner/Plugin/Framework/Search/Request/RequestCleaner.php#L100-L102. I suggest to change the handle method signature to take `$requestData` as a reference. It would also be more consistent with the methods altering the request data